packages/protocol/contracts/common/FeeBurner.sol

```diff
@@ -0,0 +1,425 @@
+import "../uniswap/interfaces/IUniswapV2FactoryMin.sol";
+import "../uniswap/interfaces/IUniswapV2PairMin.sol";
+
+contract FeeBurner is Ownable, Initializable, UsingRegistry, ICeloVersionedContract, Freezable {
+  using SafeMath for uint256;
+  using FixidityLib for FixidityLib.Fraction;
+
+  // Min units that can be burned
+  uint256 public constant MIN_BURN = 200;
+  // Historical amounts burned by this contract
+  mapping(address => uint256) public pastBurn;
+  // Max amounts that can be burned in a day for a token
+  mapping(address => uint256) public dailyBurnLimit;
+  // Max amounts that can be burned today for a token
+  mapping(address => uint256) public currentDayLimit;
+  // last day the daily limits where updated
+  uint256 public lastLimitDay;
+  // router addresses that can be set for a token
+  mapping(address => address[]) public routerAddresses;
+
+  // Max slippage a that can be accepted when burning a token
+  mapping(address => FixidityLib.Fraction) public maxSlippage;
+
+  event SoldAndBurnedToken(address token, uint256 value);
+  event DailyLimitSet(address tokenAddress, uint256 newLimit);
+  event DailyLimitHit(address token, uint256 burning);
+  event MAxSlippageSet(address token, uint256 maxSlippage);
+  event DailyLimitUpdatedDeleteMe(uint256 amount);
+  event RouterAddressSet(address token, address router);
+  event RouterAddressRemoved(address token, address router);
+  event RouterUsed(address router);
+  event ReceivedQuote(address router, uint256 quote);
+
+  /**
+   * @notice Sets initialized == true on implementation contracts
+   * @param test Set to true to skip implementation initialization
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+   * @notice Used in place of the constructor to allow the contract to be upgradable via proxy.
+   * @param _registryAddress The address of the registry core smart contract.
+   * @param tokens A list of tokens whose parameters should be set.
+   * @param newLimits A list of daily burn limits, corresponding with the same order as in the 
+      argument tokens.
+   * @param newMaxSlippages a list of max acceptable, corresponding with the same order as 
+      in the argument tokens.
+   * @param newRouters A list of routers, corresponding with the same order as in the argument 
+      tokens.
+   */
+  function initialize(
+    address _registryAddress,
+    address[] calldata tokens,
+    uint256[] calldata newLimits,
+    uint256[] calldata newMaxSlippages,
+    address[] calldata newRouters
+  ) external initializer {
+    require(tokens.length == newLimits.length, "limits lenght should match tokens'");
+    require(tokens.length == newMaxSlippages.length, "maxSlippage lenght should match tokens'");
+    require(tokens.length == newRouters.length, "maxSlippage lenght should match tokens'");
+
+    _transferOwnership(msg.sender);
+    setRegistry(_registryAddress); // TODO figureout
+
+    for (uint256 i = 0; i < tokens.length; i++) {
```

Comment:
```suggestion
    for (uint256 i = 0; i < tokens.length; i = i.add(1)) {
```

Reply:
Will leave the indexes with ++ for consistency